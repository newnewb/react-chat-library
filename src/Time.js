import PropTypes from 'prop-types';
import React from 'react';

import moment from 'moment';

import Color from './Color';
import { TIME_FORMAT } from './Constant';

export default function Time(
	{
		position,
		containerStyle,
		currentMessage,
		timeFormat,
		textStyle,
		timeTextStyle
	},
	context
) {
	return (
		<div style={[styles[position].container, containerStyle[position]]}>
			<span
				style={[
					styles[position].text,
					textStyle[position],
					timeTextStyle[position]
				]}
			>
				{moment(currentMessage.createdAt)
					.locale(context.getLocale())
					.format(timeFormat)}
			</span>
		</div>
	);
}

const containerStyle = {
	marginLeft: 10,
	marginRight: 10,
	marginBottom: 5
};

const textStyle = {
	fontSize: 10,
	backgroundColor: 'transparent',
	textAlign: 'right'
};

const styles = {
	left: {
		container: {
			...containerStyle
		},
		text: {
			color: Color.timeTextColor,
			...textStyle
		}
	},
	right: {
		container: {
			...containerStyle
		},
		text: {
			color: Color.white,
			...textStyle
		}
	}
};

Time.contextTypes = {
	getLocale: PropTypes.func
};

Time.defaultProps = {
	position: 'left',
	currentMessage: {
		createdAt: null
	},
	containerStyle: {},
	textStyle: {},
	timeFormat: TIME_FORMAT,
	timeTextStyle: {}
};

Time.propTypes = {
	position: PropTypes.oneOf(['left', 'right']),
	currentMessage: PropTypes.object,
	containerStyle: PropTypes.shape({
		left: PropTypes.object,
		right: PropTypes.object
	}),
	textStyle: PropTypes.shape({
		left: PropTypes.object,
		right: PropTypes.object
	}),
	timeFormat: PropTypes.string,
	timeTextStyle: PropTypes.shape({
		left: PropTypes.object,
		right: PropTypes.object
	})
};
