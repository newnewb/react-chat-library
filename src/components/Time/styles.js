// @ts-check

// Ours
import Colors from '../../colors';

const containerStyle = {
	marginLeft: 10,
	marginRight: 10,
	marginBottom: 5
};

const textStyle = {
	fontSize: 10,
	backgroundColor: 'transparent',
	textAlign: 'right'
};

const styles = {
	left: {
		container: {
			...containerStyle
		},
		text: {
			color: Colors.timeTextColor,
			...textStyle
		}
	},
	right: {
		container: {
			...containerStyle
		},
		text: {
			color: Colors.white,
			...textStyle
		}
	}
};

export default styles;
