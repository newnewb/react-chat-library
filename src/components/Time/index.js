// @ts-check

// Packages
const React = require('react');
const moment = require('moment');

// Ours
import { Context } from '../../context';
import { CSS } from '../../utils';
import styles from './styles';

const Time = props => {
	const { position, date, format } = props;

	return (
		<Context.Consumer>
			{ctx => (
				<div
					style={CSS([
						styles[position].container,
						props.containerStyle[position]
					])}
				>
					<span style={CSS([styles[position].text, props.textStyle[position]])}>
						{moment(date)
							.locale(ctx.locale)
							.format(format)}
					</span>
				</div>
			)}
		</Context.Consumer>
	);
};

Time.defaultProps = {
	position: 'left',
	date: null,
	containerStyle: {},
	textStyle: {},
	format: 'LT'
};

export default Time;
