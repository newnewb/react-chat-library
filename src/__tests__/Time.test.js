import React from 'react';
import renderer from 'react-test-renderer';

import Time from '../Time';

it('should render <Time /> and compare with snapshot', () => {
	const component = renderer.create(<Time />);
	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});
