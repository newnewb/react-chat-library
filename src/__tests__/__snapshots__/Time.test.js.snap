// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <Time /> and compare with snapshot 1`] = `
<div
  style={
    Object {
      "marginBottom": 5,
      "marginLeft": 10,
      "marginRight": 10,
    }
  }
>
  <span
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#aaa",
        "fontSize": 10,
        "textAlign": "right",
      }
    }
  >
    Invalid date
  </span>
</div>
`;
